math/minuit2/inc/Minuit2/FunctionGradient.h

```diff
@@ -21,30 +22,20 @@ class FunctionGradient {
 
 private:
 public:
-   explicit FunctionGradient(unsigned int n)
-      : fData(MnRefCountedPointer<BasicFunctionGradient>(new BasicFunctionGradient(n)))
-   {
-   }
+   explicit FunctionGradient(unsigned int n) : fData(std::make_shared<BasicFunctionGradient>(n)) {}
 
-   explicit FunctionGradient(const MnAlgebraicVector &grd)
-      : fData(MnRefCountedPointer<BasicFunctionGradient>(new BasicFunctionGradient(grd)))
-   {
-   }
+   // HD: this deep-copies, inconsistent to assignment?
+   explicit FunctionGradient(const MnAlgebraicVector &grd) : fData(std::make_shared<BasicFunctionGradient>(grd)) {}
```

Comment:
This is a bit weird, I copied the behavior of the original code, I am not sure if it is intentional or an oversight. If the copy ctor does a deep copy, then also assignment should make a deep copy, I think.

Generally speaking, it is quite possible that the optimizations that the original authors implemented with these shared_ptr's are not needed anymore in C++11 with move semantics.